src/shared/mesh/MeshBuilder.cpp

```diff
@@ -0,0 +1,162 @@
+  }
+  return mfem::Mesh( mesh_list.data(), mesh_list.size() );
+}
+
+MeshBuilder MeshBuilder::SquareMesh( int n_x_els, int n_y_els )
+{
+  return mfem::Mesh::MakeCartesian2D( n_x_els, n_y_els, mfem::Element::QUADRILATERAL );
+}
+
+MeshBuilder MeshBuilder::CubeMesh( int n_x_els, int n_y_els, int n_z_els )
+{
+  return mfem::Mesh::MakeCartesian3D( n_x_els, n_y_els, n_z_els, mfem::Element::HEXAHEDRON );
+}
+
+MeshBuilder MeshBuilder::HypercubeMesh( int dim, int n_els )
+{
+  switch ( dim ) {
+    case 1:
+      return mfem::Mesh::MakeCartesian1D( n_els );
+    case 2:
+      return mfem::Mesh::MakeCartesian2D( n_els, n_els, mfem::Element::QUADRILATERAL );
+    case 3:
+      return mfem::Mesh::MakeCartesian3D( n_els, n_els, n_els, mfem::Element::HEXAHEDRON );
+    default:
+      SLIC_ERROR_ROOT( "Mesh dimension must be between 1 and 3." );
+      return mfem::Mesh();
+  }
+}
+
+MeshBuilder::MeshBuilder( mfem::Mesh&& mesh ) : mesh_{ std::move( mesh ) } { mesh_.EnsureNodes(); }
+
+MeshBuilder&& MeshBuilder::translate( std::initializer_list<double> dx )
+{
+  SLIC_ERROR_ROOT_IF( static_cast<int>( dx.size() ) != mesh_.SpaceDimension(), "Invalid size for dx" );
+  auto& coords = *mesh_.GetNodes();
+  for ( int d = 0; d < mesh_.SpaceDimension(); ++d ) {
+    for ( int i = 0; i < mesh_.GetNV(); ++i ) {
+      auto vdof = coords.FESpace()->DofToVDof( i, d );
+      coords[vdof] += *( dx.begin() + d );
+    }
+  }
+  return std::move( *this );
+}
+
+MeshBuilder&& MeshBuilder::updateAttrib( int old_attrib, int new_attrib )
+{
+  for ( int i = 0; i < mesh_.GetNE(); ++i ) {
+    if ( mesh_.GetAttribute( i ) == old_attrib ) {
+      mesh_.SetAttribute( i, new_attrib );
+    }
+  }
+  auto old_attrib_idx = mesh_.attributes.Find( old_attrib );
+  if ( old_attrib_idx >= 0 ) {
+    mesh_.attributes[old_attrib_idx] = new_attrib;
+  }
+  mesh_.attributes.Sort();
+  return std::move( *this );
+}
+
+MeshBuilder&& MeshBuilder::bdrAttribInfo()
+{
+  for ( int i{ 0 }; i < mesh_.bdr_attributes.Size(); ++i ) {
+    auto current_attrib = mesh_.bdr_attributes[i];
+    auto num_elems = 0;
+    auto attrib_bbox = axom::primal::BoundingBox<double, 3>();
+    for ( int j = 0; j < mesh_.GetNBE(); ++j ) {
+      if ( mesh_.GetBdrAttribute( j ) == current_attrib ) {
+        num_elems++;
+        mfem::Array<int> bdr_dofs;
+        mesh_.GetNodes()->FESpace()->GetBdrElementDofs( j, bdr_dofs );
+        for ( int k{ 0 }; k < bdr_dofs.Size(); ++k ) {
+          axom::primal::Point<double, 3> vert;
+          for ( int d{ 0 }; d < mesh_.SpaceDimension(); ++d ) {
+            auto vdof = mesh_.GetNodes()->FESpace()->DofToVDof( bdr_dofs[k], d );
+            vert[d] = ( *mesh_.GetNodes() )[vdof];
+          }
+          attrib_bbox.addPoint( vert );
+        }
+      }
+    }
+    if ( num_elems > 0 ) {
+      std::cout << "Boundary attribute " << current_attrib << std::endl;
+      std::cout << "  Number of elements: " << num_elems << std::endl;
```

Comment:
Minor: "Number of boundary elements:"

Reply:
I think this might be OK since it says boundary on the line above?